### ISSUE TITLE
[Bug]: the comparison methods are wrong of types.TS|types.Rowid|types.Blockid 

### ISSUE THREAD

**Issue by XuPeng-SH** [matrixorigin/matrixone#18634]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

latest

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

The comparison methods are wrong regarding types.TS|types.Rowid|types.Blockid.

### Steps to Reproduce

```Markdown
drop table t12;
create table t12 (     id INT);
insert into t12 select * from generate_series(1,50000000) g;
delete from t12 where id%10=0;
delete from t12 where id%10=1;
delete from t12 where id%10=2;
Select count(*) from t12;
delete from t12 where id%10=3;
delete from t12 where id%10=4;
delete from t12 where id%10=5;
Select count(*) from t12;
delete from t12 where id%10=6;
delete from t12 where id%10=7;
delete from t12 where id%10=8;
Select count(*) from t12;
delete from t12 where id%10=9;
```

**Comment by XuPeng-SH** [comment]
rowid and blockid are fixed

**Comment by XuPeng-SH** [comment]
ts fix pr #18716
